Automattic/jetpack
# Crowdsignal: updates to get ready for Fusion

#### Changes proposed in this Pull Request:

This PR aims to fix all PHPCS warnings in the Crowdsignal / Polldaddy shortcode, so those files can be synced with WordPress.com and replace the existing shortcode there.

I made a few different changes:

- Fix PHPCS warnings
- Remove `GLOBAL_TAGS` check (we don't use that constant anywhere on WordPress.com or in Jetpack anymore).
- Remove `PADPRESS_LOADED` constant check (it was used by the Onswipe plugin, but that plugin is not in the repository anymore and we don't use the constant anywhere on WordPress.com anymore either).
- Modify the way JavaScript is enqueued. Rely on core functions (we now use `wp_register_script`, `wp_enqueue_script`, and `wp_add_inline_script` to enqueue JavaScript content on the pages).

#### Testing instructions:

**To test, you'll need to either run `yarn build` or add `define( 'SCRIPT_DEBUG', true );` to your `wp-config.php` file**

* You'll want to insert different shortcodes in your posts; some posts with only one shortcode, others with multiple shortcodes at once:
 * `[polldaddy type="iframe" survey="EB151947E5950FCF" height="auto" domain="jeherve" id="a-survey-with-branches"]`
 * `[crowdsignal type="iframe" survey="EB151947E5950FCF" height="auto" domain="jeherve" id="a-survey-with-branches"]`
 * `https://polldaddy.com/poll/7910844/`
 * `https://jeherve.survey.fm/a-survey`
 * `https://jeherve.survey.fm/a-survey-with-branches`
 * `[crowdsignal type="iframe" survey="7676FB1FF2B56CE9" height="auto" domain="jeherve" id="a-survey"]`
 * `[crowdsignal survey="7676FB1FF2B56CE9"]`
 * `[polldaddy survey="7676FB1FF2B56CE9"]`
 * `[crowdsignal poll=9541291]`
 * `[crowdsignal poll=9541291 type=slider]`
 * `[crowdsignal rating=8755352]`
* You should check that the shortcodes are displayed properly on the post's page, but also on the home page. You'll find that in some scenarios (looking at the home page on a site with a theme that supports Infinite Scroll), only a link is displayed.

@donnchawp I would appreciate your review on this PR. I don't know of all the potential embed combinations one may use, so I am afraid I may have missed one. Could you take a look?

#### Remaining question:

On WordPress.com we have an additional `site` parameter for the shortcode, introduced in r181467-wpcom. I am not sure if it is still necessary, and if we should keep it. @donnchawp Could you take a look? Would it be okay to use the short domains for all embeds now?

#### Proposed changelog entry for your changes:

* None
